webapp/src/app.tsx

```diff
@@ -1770,6 +1771,93 @@ export class ProjectView
+                (() => {
+                    for (const block of blocks) {
+                        const children = Array.from(block.children);
+                        for (const child1 of children) {
+                            if (child1.nodeName === "next") {
+                                for (const child2 of Array.from(child1.children)) {
+                                    // Grab the blocks embedded in the "next" block
+                                    blocks.push(child2);
+                                }
+                                block.removeChild(child1);
+                            }
+                        }
+                    }
+                })();
+            } catch (e) {
+                // Failed to decompile, don't propagate exception
+                console.error(`Failed to resolve blockconfig for tutorial: ${header.tutorial.tutorialName}, ${e.message}. md:${blockConfig.md}`);
+            }
+
+            for (const block of blocks) {
+                try {
+                    const entry: pxt.tutorial.TutorialBlockConfigEntry = {};
+                    const blockType = block.getAttribute("type");
+                    if (blockType === "typescript_statement") {
+                        // Decompiled to gray block
+                        throw new Error("Block config decompiled to gray block: " + Blockly.Xml.domToText(block));
+                    }
+                    if (blockType == "variables_set") {
+                        // get block id from within variables_set context
+                        const variables_set = block.children[0];
+                        const value = Array.from(variables_set.children).find(child => child.tagName === "value");
+                        const rhs = Array.from(value.children).find(child => child.tagName === "block");
+                        entry.blockId = rhs.getAttribute("type");
+                    } else {
+                        // Set block id from root type
+                        entry.blockId = blockType;
+                    }
+                    entry.xml =
+                        Blockly.Xml.domToText(block)
+                        .replace(/^<xml[^>]*>\n*/i, '')
+                        .replace(/\n*<\/xml>\n*$/i, '');
+                    blockConfig.blocks.push(entry);
+                } catch (e) {
+                    // Failed to resolve block, don't propagate exception
+                    console.error(`Failed to resolve blockconfig for tutorial: ${header.tutorial.tutorialName}. ${e.message}`);
+                }
+            }
+        };
+        const tasks: Promise<void>[] = [];
+        const stepInfo = header.tutorial.tutorialStepInfo;
+        stepInfo.forEach(step => {
+            const blockConfigs = step.blockConfigs ?? [];
+            blockConfigs.forEach(blockConfig => tasks.push(resolveBlockConfigAsync(blockConfig)));
```

Comment:
I've changed it so that it concatenates blockconfig markdown and sends it to the decompiler in one job, but it's still per tutorial step. Thinking about it from the perspective of decompiler workload, a more efficient approach would have been to make the blockconfig global to the tutorial. In fact, I'll go ahead and add support for that.